nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -266,7 +268,7 @@ class BtActionNode : public BT::ActionNodeBase
         // TODO(#1652): a work around until rcl_action interface is updated
         // if goal ids are not matched, the older goal call this callback so ignore the result
         // if matched, it must be processed (including aborted)
-        if (this->goal_handle_->get_goal_id() == result.goal_id) {
+        if (this->goal_handle_ && (this->goal_handle_->get_goal_id() == result.goal_id)) {
```

Comment:
In this case, if this callback is triggered, there must be a result so the goal handle has to be valid still, right? Did you end up actually having an issue in this callback? If not, I think this one can be removed.

If this one _does_ need to stay, then we need to have some condition to check if the  goal handle is invalid and if so, think closely about the value we should be setting for `goal_result_available_` and `result_` in that case. This would create a problem if `goal_handle_` was invalid that a result was triggered by this callback  but then the BT node's `tick()` never has a `result_` to exit https://github.com/ros-planning/navigation2/blob/e95a2299be32862cd37a25ff5571e3ceed2aaca8/nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp#L204-L212 and move onto other nodes.

Reply:
You are correct.
This check is not necessary.
I'm going to remove this one.

Reply:
According to experiment, this check is needed.